### ISSUE TITLE
[java] ReturnEmptyCollectionRatherThanNull - false negative with conditioned returns

### ISSUE THREAD

**Issue by Belle-PL** [pmd/pmd#3686]
**Affects PMD Version:6.41.0** 

**Rule:ReturnEmptyCollectionRatherThanNull**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_errorprone.html#returnemptycollectionratherthannull>

**Description:**
Hi, I think this rule should have reported warning about the following case, but it doesn't. I think this problem is caused by wrapped `return null` statemnet and it could be solved by matching.

**Code Sample demonstrating the issue:**
```java
public int[] foo() {  
  if(true) {
      return null;  // should report a warning here, but no warnings
  } else {
      return null;
  }
}
```

**Expected outcome:**

PMD should report a violation at line 3 or 5, but doesn't. This is a false-negative.

**Running PMD through:** *[CLI]*
